server/src/api/v2/event-registrations/register-for-public-event.ts

```diff
@@ -0,0 +1,67 @@
+import { Request, Response } from 'express';
+import { wrapResponse } from '../../../functions/response-wrapper';
+import { Event } from '../../../models/event.model';
+import jwt from 'jsonwebtoken';
+import { User } from '../../../models/user.model';
+import { EventRegistration } from '../../../models/event-registration.model';
+
+export async function registerForPublicEvents(req: Request, res: Response): Promise<Response> {
+    let success = true;
+    const orgId = req.params.id;
+    const eventId = req.params.eventId;
+    const currentDate = new Date();
+    // Only events, that have an end_date in the future, that are active and marked as public.
+    const eventData: Event | null = await Event.scope(
+        [
+            { method: ['onlyCurrentOrg', orgId] },
+            'public',
+            'active',
+            { method: ['notExpired', currentDate] }
+        ]
+    )
+        .findByPk(eventId)
+        .catch(() => {
+            success = false;
+            return null;
+        });
+    if (!success) {
+        return res.status(500).send(wrapResponse(false, { error: 'Database error' }));
+    }
+    if (eventData === null) {
+        return res.status(400).send(wrapResponse(false, { error: 'No public Event with given id found!' }));
+    }
+
+    const header = req.headers.authorization;
```

Comment:
Haben wir nicht eine funktion dafür? 
Wenn nein würde ich das auslagern
Warum geht das über die user is authorized middleware nicht?

Reply:
geht nicht über die middleware, weil wir darin die Org laden.
Und für die Route muss der User ja keiner Org angehören -> kann ich also nicht nutzen.
Aber ja, kann ich auslagern und dann vlt in beidem verwenden.

Reply:
Falls alle Stricke reißen, einfach ne neue middleware anlegen

Reply:
Middleware machts nicht besser und kanns leider auch nicht mehrfach verwenden, sonst muss ich noch mehr umbauen... ;)
Aber ich habs jetzt trotzdem mal da rausgezogen.